HealthcareSystem/PatientWebApp/Controllers/ExaminationController.cs

```diff
@@ -62,9 +62,13 @@ public ActionResult AdvanceSearchExaminations(ExaminationSearchDTO examinationSe
                 _examinationService.AdvancedSearch(examinationSearchDTO).ForEach(examination => examinationDTOs.Add(ExaminationMapper.ExaminationToExaminationDTO(examination)));
                 return Ok(examinationDTOs);
             }
-            catch (NotFoundException exception)
+            catch (ArgumentNullException)
             {
-                return NotFound(exception.Message);
+                return BadRequest("Patient's jmbg cannot be null.");
```

Comment:
Why does argument null exception happen here and mean this? Can you explain where in the stack this happens?

Reply:
When I changed in the service in the method for search calls a method that returns only previous examinations, NotFoundException no longer made sense, because the newly called method does not throw NotFoundException. 
Then I forwarded null to the patientJmbg and saw it being treated as ArgumentNullException, so I just changed that for that case I catch this exception.

Reply:
Alright, although it would make more sense to just validate the DTO and not call the method if it isn't correct.